### PR TITLE
first working version

### PR DESCRIPTION
**Pull request by orthagonal**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+const AssetMap = require('clientkit-asset');
+const fs = require('fs');
+const os = require('os');
+const path = require('path');
+
+class InjectRevision extends ClientkitTask {
+
+  constructor(name, options, runner) {
+    super(name, options, runner);
+    this.assetMap = new AssetMap({ pathToAssetMap: 'test/assetsMap/assets.json' });
```
Comment (jgallen23) [inline index.js]: should this be using an option?

File: index.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
+const async = require('async');
+const ClientkitTask = require('clientkit-task');
+const AssetMap = require('clientkit-asset');
```
Comment (jgallen23) [inline index.js]: I'm renaming this to clientkit-assets (plural) will release soon

File: index.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      // path to a json file containing your asset map:
+      mappingPath: 'assets.json',
+      // regexp for finding the start tag:
+      startTag: '<!-- clientkit:(.*?) -->',
+      // regex for finding the end tag:
+      endTag: '<!-- clientkit:end -->',
+      // append this to the start of every file's path:
+      uiPath: '/ui/',
+      // list of files to process:
+      files: ['index.html']
```
Comment (jgallen23) [inline index.js]: don't set this by default

File: index.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+  get defaultOptions() {
+    return {
+      // path to a json file containing your asset map:
+      mappingPath: 'assets.json',
+      // regexp for finding the start tag:
+      startTag: '<!-- clientkit:(.*?) -->',
+      // regex for finding the end tag:
+      endTag: '<!-- clientkit:end -->',
+      // append this to the start of every file's path:
+      uiPath: '/ui/',
```
Comment (jgallen23) [inline index.js]: lets have this be blank by default

File: index.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+      if (endMatch) {
+        return lines;
+      }
+    }
+    return lines;
+  }
+
+  getMiddleOfTag(originalValue, mappedValue) {
+    if (path.extname(originalValue) === '.js') {
+      return `<script type="application/javascript" src="${this.options.uiPath}${mappedValue}">`;
```
Comment (jgallen23) [inline index.js]: do you think we should do `url.resolve` when combining the two strings?

File: package.json [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    "tap-spec": "^4.1.1",
+    "tape": "^4.6.3"
+  },
+  "eslintConfig": {
+    "extends": "firstandthird"
+  },
+  "dependencies": {
+    "async": "^2.1.4",
+    "clientkit-task": "0.0.6",
+    "clientkit-asset": "0.0.1"
```
Comment (jgallen23) [inline package.json]: make this clientkit-assets

File: package.json [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+  },
+  "keywords": [],
+  "author": "Greg Allen <code@firstandthird.com> (http://firstandthird.com/)",
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/firstandthird/clientkit-inject-rev/issues"
+  },
+  "homepage": "https://github.com/firstandthird/clientkit-inject-rev#readme",
+  "devDependencies": {
+    "async": "^2.1.2",
```
Comment (jgallen23) [inline package.json]: don't need this in both places

File: package.json [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+  "scripts": {
+    "lint": "eslint .",
+    "test": "tape test/test.tape.js | tap-spec"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/firstandthird/clientkit-inject-rev.git"
+  },
+  "keywords": [],
+  "author": "Greg Allen <code@firstandthird.com> (http://firstandthird.com/)",
```
Comment (jgallen23) [inline package.json]: First and Third instead of my name

File: test/test.tape.js [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+    },
+    task2: (verifyFirstOutput, done) => {
+      const task = new InjectRevisions('revisions', {
+        files: [file]
+      });
+      task.execute(done);
+    },
+    verifySecondOutput: (task2, done) => {
+      const content = fs.readFileSync(file).toString();
+      t.equal(content.indexOf('<link rel="stylesheet" href="/ui/common-abcdefg.css">', `<head>
```
Comment (jgallen23) [inline test/test.tape.js]: can we just put an output.html in fixtures with the expected output and compare the two?

**Review by jgallen23 — CHANGES_REQUESTED** [review]

File: index.js (hunk 4 above)
Comment (orthagonal) [inline index.js]: Blank or '/'?

File: test/test.tape.js [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+
+const file = 'test/fixtures/index1.html';
+const oneOutput = fs.readFileSync('test/expected/oneOutput.html');
+const secondOutput = fs.readFileSync('test/expected/secondOutput.html');
+test('can replace the references in a file reference', (t) => {
+  t.plan(3);
+  async.autoInject({
+    task: (done) => {
+      const task = new InjectRevisions('revisions', {
+        files: [file]
```
Comment (jgallen23) [inline test/test.tape.js:31]: don't you need to pass in the path to the json?

File: index.js [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+const fs = require('fs');
+const os = require('os');
+const path = require('path');
+const url = require('url');
+
+class InjectRevision extends ClientkitTask {
+
+  constructor(name, options, runner) {
+    super(name, options, runner);
+    this.assetMap = new AssetMap({ pathToAssetMap: 'test/assetsMap/assets.json', cache: options.cache, readOnLoad: options.readOnLoad });
```
Comment (jgallen23) [inline index.js]: still hard coded

**Review by jgallen23 — CHANGES_REQUESTED** [review]

**Pull request merged**